src/plugins/intel_cpu/src/nodes/fullyconnected.h

```diff
@@ -70,6 +71,16 @@ class FullyConnected : public Node {
     void prepareParams() override;
     void executeDynamicImpl(dnnl::stream strm) override;
     bool canBeExecutedInInt8() const override;
+
+    bool canPrepInput(size_t idx) const override {
+        return idx == 1;
+    }
+
+    void prepInput(size_t idx, InputPrepType type) override {
+        OPENVINO_ASSERT(idx == 1, "Only weights input (1) can be preprocessed");
+        attrs.weightsPrepType = type;
+    }
```

Comment:
I would agree with this proposal (but probably to store this information rather outside of a node class, just some data structure inside a graph) if we had ExecutionGraph and Execution "units" as separated structures in our pipeline.
With the current code state, storing this information in a node class or graph will be just a waste of memory. There is already too much intermediate garbage inside a node object.

Reply:
My main concern here is that your proposed design doesn't force the node implementation developer set such properties explicitly. The implementation doesn't have to clearly define which inputs are preprocessed by the node itself, and the developer must somehow know about this feature to optimize constants processing. Such a design flaw isn't harmful in itself, as leaving a default behavior doesn't break program correctness. But, it looks like a parameter of the operation semantics, which is a property, and may be request to be set explicitly. That would enforce developers apply such optimization for future node implementations.

Reply:
I don't see a big difference in those cases to be honest. I agree that this could be implemented as part of some Node / Graph configuration parameter. But in this case, this parameter could also have a default value. Or it might not have it.
If let's say we decide to not implement a default value for this parameter to force a developer to explicitly specify this parameter for every node, then it would require setting it to some default value in scope of every node.
With the current approach this is basically the same. We can make those virtual methods pure and require to explicitly override them for every node.
One of the examples can be current `needPrepareParams` method. It has its default implementation, and by default, developer might not know that it can override it to achieve better performance / logic. This is absolutely normal way of working.
And, again, with the current state of the source code the approach with virtual functions is better, because it avoids storing temporary data in the Node object.